software/known-issues.md

```diff
@@ -123,3 +131,107 @@ KNOWN_CURSORS=["HORIZON"]
 5. call "newdb" on the core instance to reset core's database
 6. start up stellar-core normally, wait for it to be in "Synced!" state
 7. start up Horizon, it should be able to see that core has data to ingest
+
+# Backfilling history
+
+You have a core instance that works properly (close ledgers), but you realize
+that for some reason its history is incomplete, in the sense that the oldest
+ledger available for Horizon to import is not old enough.
+
+Reasons that can lead to this include some of the problems that can lead to
+[Gaps Detected](#gaps-detected).
+
+## Check your configuration
+
+Follow the steps in [mismatched core/Horizon configurations](#mismatched-corehorizon-configurations)
+
+In particular, if the reason you have missing data is because of
+garbage collection, you want to make sure that it doesn't happen again in the
+future!
+
+## Recovery process
+
+Important, before attempting any recovery process:
+* make sure that your configuration is correct (and if not adjust it)
+* **BACKUP** all data on your core node
+
+### Reset everything
+
+The easiest way to recover is to [reset everything](#reset-everything).
+
+This will take some (time for machines) but doesn't require manual intervention.
+
+### Manually backfill history
+
+The general idea behind this method is to construct the missing data on a
+separate stellar-core instance, followed by an "import" into the real instance.
+
+Note that **Horizon does not support backfilling**: it will have to reingest all
+data from core.
```

Comment:
There's `horizon db backfill [COUNT]` command. So I think that when you download `LOW_LEDGER .. ORIGINAL_LOW_LEDGER` ledgers using a new stellar-core instance you could then switch Horizon to the new Core instance and use `backfill` command.

Reply:
@bartekn When I ran horizon db backfill, I couldn't get it to load the starting point correct, which is defined in this line:
`start := ledger.CurrentState().HistoryElder` (system.go:20)
It would instead make the start = 0 and I couldn't get it to backfill.

Maybe it would be useful if we can extend the `horizon db reingest` command to backfill a range that the user can specify. What do you think about that?

Reply:
You are experiencing this because either your history DB is empty or a bug in Horizon (`ledger` state is not updated before it's first called in `system.go:20` as it's part of the ticking service). Can you file a bug in `stellar/go`? The fix is rather easy (update the `ledger` state before starting any Horizon CLI command).

Reply:
Feel free to open a PR to document backfilling in Horizon to complement this guide: that said I'd like to keep this guide as simple as possible; I imagine that using this feature comes with a bunch of "gotchas".

Reply:
OK! FYI it has been fixed: https://github.com/stellar/go/pull/723.